### DISCUSSION TITLE
Building for ESP32C3 with hardware RTC 32.768KHz crystal

### DISCUSSION THREAD

**Discussion by feeley** [micropython/micropython#11118] [ESP32]
Is there a way to build micropython on ESP32C3 so that the RTC is configured as the "external low-speed (32 kHz) crystal clock" (on pins GPIO0 and GPIO1)?

**Comment by 2dof** [comment]
based on -C3 APIdescription You can use External [32kHz Crystal - optional (XTAL32K)](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/peripherals/clk_tree.html), but with micropython  You need change  sdkconfig file , set CONFIG_ESP32_RTC_CLOCK_SOURCE_EXTERNAL_CRYSTAL=1 and build custom firmware. [issues/3764](https://github.com/micropython/micropython/issues/3764)


 [see also user guide section: 6.2 Clock](https://www.espressif.com/sites/default/files/documentation/esp32-c3_technical_reference_manual_en.pdf)  - based on user guide  external 32.7 XTAL is used for  "low-power devices, such as RTC module and low-power peripherals"  but not for a CPU clock source.

**Comment by feeley** [comment]
Thanks! I will give it a try.
